### PR TITLE
Example for tag pairs and code regeneration

### PR DESCRIPTION
**Pull request by verdverm**

### CODE REVIEW

**Comment by raphael** [comment]
Looks good! can you update the PR so that it only applies to controllers? For example the README talks about comments in `main` but these don't exist as the corresponding PR wasn't merged. Thanks!

**Comment by verdverm** [comment]
I believe the README is free of `main` tag references now

File: code_regen/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+goa Code Regeneration
+=====================
+
+Note: This depends on a
+[PR in progress](https://github.com/goadesign/goa/pull/593) for changing TDB to the tag pairs.
+
+---
+
+This example shows how to regenerate your apps code
```
Comment (raphael) [inline code_regen/README.md]: might be good to specify a bit more about the context here so people don't think this is required to update the design. This script is just a helper to re-generate a controller so you can quickly take advantage of the scaffold generated for its handlers. at the end of the day the controller code is user code and should be treated as such.

File: code_regen/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+`extra` will be appended to the end of the file
+and allows extra functions to be added to the file.
+
+
+```Go
+package main
+
+import (
+  // import start_implement
+  "fmt"
```
Comment (raphael) [inline code_regen/README.md]: are the import tags really needed? imports change rarely and when they do most people use `goimport` so it doesn't matter anyway. Would be good to keep the script to a minimum.

File: code_regen/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+func hello() {
+  fmt.Println("hello")
+}
+// extra end_implement
+
+```
+
+The Makefile also wraps the goagen workflow.
+
+- `make gen` : goagen
```
Comment (raphael) [inline code_regen/README.md]: for this example please keep only one target for the makefile (the default target) that simply does the regen. Again trying to keep things as simple as possible.

**Comment by raphael** [comment]
Thank you! left a few comments in the commit that boil down to two things:

1. Let's make sure people understand that this is just a little helper to be able to quickly take advantage of the scaffold generated for the controllers implementation. In general goa has a clear strategy to enable the design lifecycle and this is just for convenience.

2. Please keep the makefile and scripts as simple as possible and simply to address the use case described in 1.

Thanks for doing this!

**Comment by raphael** [comment]
Thank you, looks great!

**Pull request merged**